### DISCUSSION TITLE
More Generic QSPI Drivers

### DISCUSSION THREAD

**Discussion by JasteD** [NXPmicro/mcux-sdk#128] [Ideas]
Dear Community,

I'm currently evaluating the NXP RT1176 for a new application. Therefor it is necessary to have a QSPI interface to control a sensor. As the SDK currently only provides a QSPI interface which is dedicated for memory use cases, I want to ask if there is the possibility, that there will be a release of a more generic QSPI Driver which supports the following QSPI functions for read/write operations: Instructions Phase (multiple line modes like single line, dual line, quad line), Address Phase (multiple line modes), Alternative Bytes Phase (multiple line modes), Dummy Cycles, Data Phase (multiple line modes) and configure the idle state of DIO0-3 and CLK. 

I would be happy to start a discussion here and to get feedback from NXP Support if it is realistic to add this feature to the SDK.

Best regards, 
Jurni_JasteD

**Comment by mcuxsusan** [comment]
Hi, do you mean the FlexSPI driver? As far as I know, you need to set up LUT for a FlexSPI transaction, and in the LUT you should be able to customize Instructions Phase (multiple line modes like single line, dual line, quad line), Address Phase (multiple line modes), Alternative Bytes Phase (multiple line modes), Dummy Cycles, Data Phase (multiple line modes).
